src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+     *
+     * @return true if random generator is statistical.
+     */
+    public boolean isStatistical() {
+        return !(Boolean)getProperty(RandomGeneratorProperty.IS_STOCHASTIC, true);
+    }
+
+    /**
+     * Return true if random generator is stochastic.
+     *
+     * @return true if random generator is stochastic.
+     */
+    public boolean isStochastic() {
+        return (Boolean)getProperty(RandomGeneratorProperty.IS_STOCHASTIC, false);
+    }
+
+    /**
+     * Return true if random generator is generated by hardware.
+     *
+     * @return true if random generator is generated by hardware.
+     */
+    public boolean isHardware() {
+        return (Boolean)getProperty(RandomGeneratorProperty.IS_HARDWARE, false);
+    }
+
+    /**
+     * Return true if random generator is arbitrarily jumpable.
+     *
+     * @return true if random generator is arbitrarily jumpable.
+     */
+    public boolean isArbitrarilyJumpable() {
+        return isSubclass(ArbitrarilyJumpableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator is jumpable.
+     *
+     * @return true if random generator is jumpable.
+     */
+    public boolean isJumpable() {
+        return isSubclass(JumpableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator is leapable.
+     *
+     * @return true if random generator is leapable.
+     */
+    public boolean isLeapable() {
+        return isSubclass(LeapableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator is splittable.
+     *
+     * @return true if random generator is splittable.
+     */
+    public boolean isSplittable() {
+        return isSubclass(SplittableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator is streamable.
+     *
+     * @return true if random generator is streamable.
+     */
+    public boolean isStreamable() {
+        return isSubclass(StreamableGenerator.class);
+    }
+
+    /**
+     * Create an instance of {@link RandomGenerator} based on algorithm chosen.
+     *
+     * @return new in instance of {@link RandomGenerator}.
+     *
+     */
+    public T create() {
+        try {
+            ensureConstructors();
+            return ctor.newInstance();
+        } catch (Exception ex) {
+            // Should never happen.
+            throw new IllegalStateException("Random algorithm " + name() + " is missing a default constructor");
+        }
+    }
+
+    /**
+     * Create an instance of {@link RandomGenerator} based on algorithm chosen
+     * providing a starting long seed. If long seed is not supported by an
+     * algorithm then the no argument form of create is used.
+     *
+     * @param seed long random seed value.
+     *
+     * @return new in instance of {@link RandomGenerator}.
+     */
+    public T create(long seed) {
+        try {
+            ensureConstructors();
+            return ctorLong.newInstance(seed);
+        } catch (Exception ex) {
```

Comment:
this one is very dubious because the result in an exception is thrown is a random generator with the wrong seed

Reply:
This is explained in the docs.